Microsoft.Research/Contracts/System.Core.Contracts/System.Linq.Expressions.Expression.cs

```diff
@@ -5566,8 +5566,14 @@ public sealed class Expression<TDelegate>
   {
     private Expression() { }
 
-    // public TDelegate Compile();
+#if !SILVERLIGHT
+        public TDelegate Compile()
+        {
+            Contract.Ensures(!ReferenceEquals(Contract.Result<TDelegate>(), null));

```

Comment:
Because there is no constraint that TDelegate is a class, so for the compile it could be a value type, too - and you cant do value_type != null